explorerlib/index.go

```diff
@@ -45,17 +55,33 @@ func (index *Index) FetchResponse(queryExpression string, numberOfResults int) (
 	})
 }
 
-func (index *Index) BuildGoodQueries(wordCountsByLanguage map[string]WordCounts, numberOfQueryByLanguage int, averageNumberOfWords int) (map[string][]string, error) {
+func (index *Index) BuildGoodQueries(wordCountsByLanguage map[string]WordCounts, numberOfQueryByLanguage int, averageNumberOfWords int, minTime time.Duration) (map[string][]string, error) {
+
+	numberOfActiveBot++
+	throttle = (minTime * time.Millisecond ) * time.Duration(numberOfActiveBot)
+	scenariolib.Info.Printf("Throttled at : %v",throttle)
+
 	queriesInLanguage := make(map[string][]string)
-	scenariolib.Info.Print("Building queries and calling the index to validate that they return results")
+	scenariolib.Info.Print("Building queries and calling the index to validate that they return results \n")
+
 	for language, wordCounts := range wordCountsByLanguage {
 		words := []string{}
+
 		for i := 0; i < numberOfQueryByLanguage; {
 			word := wordCounts.PickExpNWords(averageNumberOfWords)
+
+			t2 = time.Now()
```

Comment:
In this order, dt2 includes the sleep of the previous step. I propose:
``` go
dt2 = time.Since(t2)
if dt2 < throttle{
     time.Sleep(throttle - dt2)
}
t2 = time.Now()
response, err := index.FetchResponse(word, 10)
 ```
And initialize t2 at time.Now() before the for loop.